src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/ReviewCodeForSqlInjectionVulnerabilitiesTests.cs

```diff
@@ -622,6 +624,40 @@ protected void Page_Load(object sender, EventArgs e)
                 GetCSharpResultAt(15, 37, 15, 52, "SqlCommand.SqlCommand(string cmdText)", "void WebForm.Page_Load(object sender, EventArgs e)", "string HttpRequest.this[string key]", "void WebForm.Page_Load(object sender, EventArgs e)"));
         }
 
+        [Fact]
+        public async Task HttpRequest_Form_Method_GenericsSink_Diagnostic()
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetFramework.Net472.Default
+                    .AddAssemblies(ImmutableArray.Create("System.Web", "System.Web.Extensions"))
+                    .AddPackages(ImmutableArray.Create(new PackageIdentity("EntityFramework", "6.4.4"))),
+                TestState =
+                {
+                    Sources = { @"
+namespace VulnerableWebApp
+{
+    using System;
+    using System.Data.Entity;
+    using System.Web.UI;
+
+    public partial class WebForm : System.Web.UI.Page
+    {
+        protected void Page_Load(object sender, EventArgs e)
+        {
+            string input = Request.Form.Get(""in"");
+            new DbContext(""connectionString"").Set<Object>().SqlQuery(input, null);
+        }
+     }
+}
+" }
+                },
+            };
+
+            csharpTest.ExpectedDiagnostics.AddRange(new[] { GetCSharpResultAt(13, 13, 12, 28, "DbSqlQuery<object> DbSet<object>.SqlQuery(string sql, params object[] parameters)", "void WebForm.Page_Load(object sender, EventArgs e)", "NameValueCollection HttpRequest.Form", "void WebForm.Page_Load(object sender, EventArgs e)") });
```

Comment:
Optional too but you could have the expected diagnostics declared directly in the initializer (that's what is generally done when there is no condition to adding the diagnostics).

Reply:
The property is read-only. Since this is optional I'll rather skip it.